### ISSUE TITLE
go: Error: Failure while executing ldd

### ISSUE THREAD

**Issue by timotheecour** [Linuxbrew/homebrew-core#2578]
@sjackman 

- [x] Confirmed this is a problem with Linuxbrew and not Homebrew. Problems that affect both Homebrew and Linuxbrew must be reported in the appropriate Homebrew repository https://github.com/Homebrew
- [x] Confirmed this is a problem with only one, specific formula and not Linuxbrew/brew? If it's a general Linuxbrew/brew problem please file this issue at https://github.com/Linuxbrew/brew/issues/new
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

from a fresh brew install:
1st bug:
```
brew install go
==> go build
Error: patchelf must be installed: brew install patchelf
```

then after fixing this bug, i get 2nd bug:
```
brew install patchelf
#ok
brew install go

==> Downloading https://storage.googleapis.com/golang/go1.8.1.src.tar.gz
Already downloaded: /home/timothee/.cache/Homebrew/go-1.8.1.tar.gz
==> Downloading https://storage.googleapis.com/golang/go1.7.linux-amd64.tar.gz
Already downloaded: /home/timothee/.cache/Homebrew/go--gobootstrap-1.7.tar.gz
==> ./make.bash --no-clean
==> /home/timothee_local/temp/brew3/Cellar/go/1.8.1/bin/go install -race std
==> Cloning https://go.googlesource.com/tools.git
Updating /home/timothee/.cache/Homebrew/go--gotools--git
==> Checking out branch release-branch.go1.8
==> go build
Error: Failure while executing: ["ldd", "bew/Cellar/go/1.8.1/libexec/src/debug/elf/testdata/gcc-386-freebsd-exec"]

```

NOTE: brew gist-logs go never completes (waited 30mn)

**Comment by maxim-belkin** [comment]
@timotheecour
Could you please try the following and report if you get any further
```
cd $(brew --repo)
git fetch origin
git checkout master
brew install go
```

**Comment by sjackman** [comment]
Hi, @timotheecour. This is a bug in `brew`.  `brew install go` fails because `brew linkage --test go` fails because`patchelf --print-needed` fails for `go` executables. The formula `guile` also has this issue. `go` executables look really weird.
```
❯❯❯ patchelf --print-soname /home/sjackman/.linuxbrew/Cellar/guile/2.2.2/lib/guile/2.2/ccache/ice-9/and-let-star.go
cannot find section .dynstr
❯❯❯ readelf -d /home/sjackman/.linuxbrew/Cellar/guile/2.2.2/lib/guile/2.2/ccache/ice-9/and-let-star.go
Dynamic section at offset 0x2f30 contains 7 entries:
  Tag        Type                         Name/Value
 0x0000000037146003 (<unknown>: 37146003) 0x2020041
 0x0000000037146002 (<unknown>: 37146002) 0xe8
 0x0000000037146000 (<unknown>: 37146000) 0x10000
 0x0000000037146001 (<unknown>: 37146001) 0x18b0
 0x000000000000000c (INIT)               0xbf8
 0x0000000037146004 (<unknown>: 37146004) 0x2e70
 0x0000000000000000 (NULL)               0x0
```
`brew` needs to detect this situation and either skip these wacky executables or ignore the `patchelf` error.

**Comment by sjackman** [comment]
I've opened PR https://github.com/Linuxbrew/brew/pull/365 to address this issue.

**Comment by sjackman** [comment]
I've merged PR #365, which should resolve this issue.
